rules/utils/needs-semicolon.js

```diff
@@ -43,12 +43,44 @@ function needsSemicolon(tokenBefore, sourceCode, code) {
 	}
 
 	const {type, value, range} = tokenBefore;
+	const lastBlockNode = sourceCode.getNodeByRangeIndex(range[0]);
 	if (type === 'Punctuator') {
 		if (value === ';') {
 			return false;
 		}
 
-		if (value === ']' || value === ')') {
+		if (value === ']') {
```

Comment:
This part not related to this PR, but found during fix this.

```js
foo.forEach(_ => {
	if (true) return {};
})
```

was fixed to 

```js
foo.forEach(_ => {
	if (true) ;({}); continue;
              ^ Should not insert
})
```

I'll send another PR to add tests cover this part.